### ISSUE TITLE
RCS1215 false positive

### ISSUE THREAD

**Issue by mikernet** [JosefPihrt/Roslynator#1734]
`RCS1215` incorrectly triggers here for some reason:

<img width="1108" height="216" alt="Image" src="https://github.com/user-attachments/assets/b679beec-81c6-4760-9b22-a1de31f936ae" />

If the `||` clauses are reversed then it does not show the warning:

```cs
if ((SelectedIconGroup is not null && !FilteredIconGroups.Contains(SelectedIconGroup)) || SelectedIconGroup is null)
```
